install.sh

```diff
@@ -122,30 +122,103 @@ install_python3_and_sqlite3_from_source_with_yum() {
   cd "$CURRENT_WD"
 }
 
+find_python() {
+  set +e
+  unset BEST_VERSION
+  for V in 310 3.10 39 3.9 38 3.8 37 3.7 3; do
+    if command -v python$V >/dev/null; then
+      if [ "$BEST_VERSION" = "" ]; then
+        BEST_VERSION=$V
+        if [ "$BEST_VERSION" = "3" ]; then
+          PY3_VERSION=$(python$BEST_VERSION --version | cut -d ' ' -f2)
+          if [[ "$PY3_VERSION" =~ 3.11.* ]]; then
+            echo "Chia requires Python version < 3.11.0" >&2
+            echo "Current Python version = $PY3_VERSION" >&2
+            # If Arch, direct to Arch Wiki
+            if type pacman >/dev/null 2>&1 && [ -f "/etc/arch-release" ]; then
+              echo "Please see https://wiki.archlinux.org/title/python#Old_versions for support." >&2
+            fi
+            exit 1
+          fi
+        fi
+      fi
+    fi
+  done
+  echo $BEST_VERSION
+  set -e
+}
+
+INSTALL_PYTHON_PATH=
+SQLITE_VERSION=
+SQLITE_MAJOR_VER=
+SQLITE_MINOR_VER=
+OPENSSL_VERSION_STRING=
+OPENSSL_VERSION_INT=
+
+if [ -z "$INSTALL_PYTHON_VERSION" ]; then
+  INSTALL_PYTHON_VERSION=$(find_python)
+fi
+
+if [ -n "$INSTALL_PYTHON_VERSION" ]; then
+  INSTALL_PYTHON_PATH=python${INSTALL_PYTHON_VERSION}
```

Comment:
Should this have the 3.7 default from below?  I'll note that `Install.ps1` actually tries each supported version from newest to oldest.  But that should probably be done separately if of interest.

Reply:
Code updated.